### PR TITLE
Allowing download artifacts from other workflows

### PR DESCRIPTION
**Pull request by germa89**
As the title.

We should probably give many options for filtering the artifacts.

I'm using the default ``actions/download-artifact`` action unless one of the arguments is used, but I'm happy to drop it.

### CODE REVIEW

File: doc-deploy-dev/action.yml [hunk 1]
```diff
@@ -15,6 +15,39 @@ inputs:
...
+    required: false
+  run_number:
+    description: Workflow run number
+    required: false
+  name:
+    description: Artifact name (download all artifacts if not specified)
+    required: false
+  dry_run:
+    description: Check the existence of artifact(s) without downloading
+    required: false
```
Comment (germa89) [inline doc-deploy-dev/action.yml:50]: Copy/pasted from the ``https://github.com/pyansys/actions/pull/121`` args

File: doc-deploy-dev/action.yml [hunk 2]
```diff
@@ -34,11 +67,29 @@ runs:
       shell: bash
       run: rm -rf dev/ && mkdir dev/
 
-    - name: "Download the HTML documentation artifact"
+    - name: "Download the HTML documentation artifact from the same workflow"
       uses: actions/download-artifact@v3
+      if: ${{ inputs.workflow == "" }}
+      with:
+        name: ${{ inputs.doc-artifact-name }}
+        path: dev
```
Comment (germa89) [inline doc-deploy-dev/action.yml:75]: This branching can be deleted if we are happy with using ``dawidd6/action-download-artifact@v2`` for everything... I think we can be confident enough we should not mess it up.

**Comment by germa89** [comment]
Interesting read:

https://www.legitsecurity.com/blog/github-actions-that-open-the-door-to-cicd-pipeline-attacks

**Comment by germa89** [comment]
https://github.com/dawidd6/action-download-artifact

**Comment by germa89** [comment]
Replace #120

File: doc-deploy-dev/action.yml (hunk 1 above)
Comment (jorgepiloto) [inline doc-deploy-dev/action.yml:50]: Please, make sure that each input has the following fields:

* Description: "...." (Use single quotes
* Default (if required): default-value
* Required: true/false
* Type: [boolean, string...]

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

File: doc-deploy-dev/action.yml (hunk 2 above)
Comment (germa89) [reply doc-deploy-dev/action.yml:75]: @jorgepiloto  What about this?

File: doc-deploy-dev/action.yml (hunk 1 above)
Comment (germa89) [reply doc-deploy-dev/action.yml]: This is now resolved.

File: doc-deploy-dev/action.yml (hunk 2 above)
Comment (germa89) [reply doc-deploy-dev/action.yml]: I'm just going to use that action for everything.

File: doc-deploy-dev/action.yml [hunk 3]
```diff
@@ -15,6 +15,48 @@ inputs:
...
+  run_number:
+    description: 'Workflow run number'
+    required: false
+    type: string
+  name:
+    description: 'Artifact name (download all artifacts if not specified)'
+    required: false
+    type: string
+  dry_run:
+    description: 'Check the existence of artifact(s) without downloading'
```
Comment (MaxJPRey) [inline doc-deploy-dev/action.yml:57]: This could be very useful. 👍

File: doc-deploy-dev/action.yml [hunk 4]
```diff
@@ -15,6 +15,48 @@ inputs:
     description: "Required token for documentation deployment."
     required: true
     type: string
+  workflow:
+    description: 'Workflow name from where the artifacts are downloaded. By default, the same running workflow'
```
Comment (MaxJPRey) [inline doc-deploy-dev/action.yml:19]:
```suggestion
    description: 'Workflow name from where the artifacts are downloaded. By default, the same running workflow.'
```

**Review by MaxJPRey — APPROVED** [review]
I needed something very similar few days ago. Thanks @germa89 for implementing it.

**Comment by germa89** [comment]
> I needed something very similar few days ago. Thanks @germa89 for implementing it.

It is my pleasure @MaxJPRey !! :)

**Review by jorgepiloto — APPROVED** [review]
Awesome, @germa89!

**Pull request merged**